### PR TITLE
Switch from whacko to jsdom, fix #301

### LINKED ISSUE #301: Switch from whacko to jsdom

### PR DESCRIPTION
**Pull request by jennyliang220**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -52,25 +53,23 @@ app.get("/", function (req, res) {
...
-                    .replace(":", "")
-                    .replace("published ", "")
-                    .trim();
-                    $dd = $dt.next();
-                    if (txt === "this version") {
-                        thisURI = $dd.find('a').attr('href');
-                    }
+            if (dl) {
+                dl.querySelectorAll('dt').forEach(dt => {
+                    var txt = (dt.innerText || dt.textContent)
```
Comment (marcoscaceres) [inline server.js:62]: Maybe dt.textContent first? .innerText should probably not be needed.

**Comment by deniak** [comment]
@marcoscaceres it's just a long standing issue. `whacko` has been abandoned for a few years now and we are simply running late on replacing it.

File: package.json [hunk 2]
```diff
@@ -6,9 +6,10 @@
   "dependencies": {
     "express": "4.17.1",
     "express-fileupload": "1.1.6",
+    "jsdom": "16.2.2",
+    "nodemon": "^2.0.4",
```
Comment (deniak) [inline package.json:10]: move `nodemon` to the `devDependencies`

**Comment by deniak** [comment]
@jennyliang220 I think the travis build is failing because of the npm hook `prepublish`. I can't reproduce the failure locally and the packages are correctly installed before `snyk protect` is called but looking at the travis log, it does seem that the dependencies aren't installed.
The following warning suggest to use a different hook as `prepublish` is deprecated:
```
npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.
```

I haven't tried it myself but it's worth a try.

File: server.js (hunk 1 above)
Comment (deniak) [reply server.js]: resolved in https://github.com/w3c/spec-generator/pull/306/commits/b881e684b6d592f409f1c858fd688b9e59e7a30c

**Pull request merged**